### PR TITLE
[16.0][MIG] jsonifier

### PR DESCRIPTION
**Pull request by sebastienbeau**

### CODE REVIEW

**Comment by sebastienbeau** [comment]
@PierrickBrun @simahawk @lmignon can you review so we can merge it ?

**Review by PierrickBrun — APPROVED** [review]
code review only

**Review by lmignon — APPROVED** [review]

File: jsonifier/models/models.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        if one:
+            self.ensure_one()
+        if isinstance(parser, list):
+            parser = convert_simple_to_full_parser(parser)
+        resolver = parser.get("resolver")
+
+        results = [{} for record in self]
+        parsers = {False: parser["fields"]} if "fields" in parser else parser["langs"]
+        records = self
+        if "jsonifier__date_user_tz" not in self.env.context:
```
Comment (simahawk) [inline jsonifier/models/models.py]: can you take care of this?

**Review by nimarosa — APPROVED** [review]

File: jsonifier/models/models.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            # TODO: backward compat flag for v < 15.0
+            # Dates must always be UTC and the client/consumer of this data
+            # should deal w/ the format as preferred.
+            # Drop this flag for v15!
+            records = records.with_context(jsonifier__date_user_tz=True)
```
Comment (lmignon) [inline jsonifier/models/models.py]: @sebastienbeau  `jsonifier__date_user_tz` must be removed from the code and all the related code.

**Review by lmignon — CHANGES_REQUESTED** [review]

**Comment by sebastienbeau** [comment]
@lmignon @simahawk it's ok for you ?

File: jsonifier/tests/test_get_parser.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
+# Copyright 2017 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import mock
```
Comment (lmignon) [inline jsonifier/tests/test_get_parser.py]:
```suggestion
from unittest import mock
```

**Review by lmignon — CHANGES_REQUESTED** [review]
A little change to make the test :green_heart:

File: jsonifier/tests/test_get_parser.py (hunk 3 above)
Comment (sebastienbeau) [reply jsonifier/tests/test_get_parser.py]: done

**Comment by sebastienbeau** [comment]
I am fixing the test

**Review by lmignon — APPROVED** [review]

**Pull request merged**